src/lavinmq/client/channel.cr

```diff
@@ -384,6 +384,9 @@ module LavinMQ
             @client.vhost.event_tick(EventType::ClientGet)
             ok = q.basic_get(frame.no_ack) do |env|
               delivery_tag = next_delivery_tag(q, env.segment_position, frame.no_ack, nil)
+              unless frame.no_ack # track unacked messages
+                q.basic_get_unacked << UnackedMessage.new(delivery_tag, "", @client.channels.first_value.name, RoughTime.monotonic)
```

Comment:
Why not?
```suggestion
                q.basic_get_unacked << UnackedMessage.new(delivery_tag, "", @name, RoughTime.monotonic)
```

but in addition to that i think `UnackedMessage` should include a reference to the actual `Channel`. 
```suggestion
                q.basic_get_unacked << UnackedMessage.new(delivery_tag, "", self, RoughTime.monotonic)
```
And no basic-get message ever has a consumer tag so: 
```suggestion
                q.basic_get_unacked << UnackedMessage.new(self, delivery_tag RoughTime.monotonic)
```

Reply:
changed!